common/src/main/java/org/terracotta/angela/common/util/RetryUtils.java

```diff
@@ -63,11 +68,17 @@ public static boolean waitFor(Callable<Boolean> condition, int maxWaitTimeMillis
     return success;
   }
 
+  private static void sleep(long currRetryTime) {
+    try {
+      Thread.sleep(currRetryTime);
+    } catch (InterruptedException e) {
+      logger.debug("Thread interrupted. Ignoring the exception since we need to retry");
```

Comment:
missing `Thread.currentThread().interrupt()` so that the interrupt flag gets reset back for the caller thread to know there has been a thread interruption

Reply:
All this checking and retrying happens in the same thread. To me, it doesn't matter if the thread was interrupted or not, we're going to retry. Setting the interrupt flag back doesn't seem to be helping here..

Reply:
If the user aasks for thread interruption, you shouldn't retry at all... In the while loop, you should have something like:
while (... && !Thread.currentThread().isINterrupted()) {
}
and if the thread has been interrupted because the user wants to "stop" the test for example, then it will allow a shorter shutdown.

Reply:
https://codeahoy.com/java/How-To-Stop-Threads-Safely/

Reply:
Good point about the user-initiated test termination.

Updated the PR with review comments. Please check and merge if it looks good.